src/main/java/refinedstorage/tile/TileDestructor.java

```diff
@@ -99,20 +99,24 @@ public void updateNode() {
             BlockPos front = pos.offset(getDirection());
 
             if (pickupItem && type == IType.ITEMS) {
-                List<Entity> droppedItem = new ArrayList<>();
+                List<Entity> droppedItems = new ArrayList<>();
 
                 Chunk chunk = worldObj.getChunkFromBlockCoords(front);
-                chunk.getEntitiesWithinAABBForEntity(null, new AxisAlignedBB(front), droppedItem, null);
-
-                if (droppedItem.size() > 0) {
-                    for (Entity item : droppedItem) {
-                        if (item instanceof EntityItem) {
-                            ItemStack remainder = network.insertItem(((EntityItem) item).getEntityItem().copy(), ((EntityItem) item).getEntityItem().stackSize, false);
-                            if (remainder != null) {
-                                InventoryHelper.spawnItemStack(worldObj, front.getX(), front.getY(), front.getZ(), remainder);
+                chunk.getEntitiesWithinAABBForEntity(null, new AxisAlignedBB(front), droppedItems, null);
+
+                for (Entity item : droppedItems) {
+                    if (item instanceof EntityItem) {
+                        if (IFilterable.canTake(itemFilters, mode, compare, ((EntityItem) item).getEntityItem())) {
+                            int originalSize = ((EntityItem) item).getEntityItem().stackSize;
```

Comment:
Also, you seem to be using ((EntityItem) item).getEntityItem() a lot, please extract that in a variable